src/Rings/mpoly-localizations.jl

```diff
@@ -143,13 +143,24 @@ function simplify!(S::MPolyPowersOfElement)
   new_denom = Vector{elem_type(R)}()
   for d in denominators(S)
     for a in factor(d)
-      push!(new_denom, a[1])
+      !(a in new_denom) && push!(new_denom, a[1])
     end
   end
   S.a = new_denom
   return S
 end
 
+function simplify(S::MPolyPowersOfElement)
+  R = ambient_ring(S)
+  new_denom = Vector{elem_type(R)}()
+  for d in denominators(S)
+    for a in factor(d)
+      !(a in new_denom) && push!(new_denom, a[1])
```

Comment:
```suggestion
      !(a[1] in new_denom) && push!(new_denom, a[1])
```
?

Reply:
Oops... Yes, this should probably be fixed. Thanks.